client/asset/dcr/externaltx.go

```diff
@@ -328,7 +328,7 @@ func (dcr *ExchangeWallet) isOutputSpent(ctx context.Context, output *outputSpen
 	nextScanHeight := lastScannedHeight + 1
 
 	bestBlock := dcr.cachedBestBlock()
-	if nextScanHeight >= bestBlock.height {
+	if nextScanHeight > bestBlock.height {
```

Comment:
This was awkward during testing and I think it was incorrect. It skipped the best block, which has one confirm, so transactions were not searched there. I do not see an comments about a reason for this.

Reply:
@itswisdomagain See if this is correct please.

Reply:
Yeah, looks correct per the comment below this if statement. I think that was supposed to be and I think it should be changed to `if lastScannedHeight >= bestBlock.height`.

The `nextScanHeight := lastScannedHeight + 1` should be moved to after this if statement.

The idea, per the comment inside the if statement, is ...
```
if lastScannedHeight == bestBlock.height {
	// this is fine, just return, no more blocks to scan
	return ...
}
if lastScannedHeight > bestBlock.height {
	// not fine, how did it scan a height above the best height?
	// log a warning, should never happen anyways
	// return too, since there're obviously no more blocks to scan
	return ...
}
```

Reply:
This look right? https://github.com/decred/dcrdex/compare/db62034bae674721bf653a8b666faf8e1d4b5ad3..53bd2f1384b8ba7c09ac0ff40302d7c1085de111